RFC-005-SWAP-Bitcoin-basic.md

```diff
@@ -0,0 +1,174 @@
+
+### Deployment
+
+The party who is required to deploy the Bitcoin HTLC (the refunder) compiles the contract as described in the previous section.
+Then, they MUST send a transaction to the Bitcoin Blockchain with an output with value exactly equal to the quantity of satoshi specified by `asset` and a Pay-To-Witness-Script-Hash (P2WSH) `scriptPubKey` derived from the contract. See [BIP141](https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#specification)
+for how to construct the `scriptPubkey`.
+
+The redeeming party should likewise compile the contract script and wait until the above transaction appears in the Bitcoin blockchain with enough confirmations such that they consider it permanent. They should do this by watching the Blockchain for a transaction with an output matching the `scriptPubkey` and having the required value.
+
+#### Redeem
+
+To redeem from the HTLC, the redeemer should submit a transaction to the blockchain which spends the P2WSH output.
+The redeemer can use following witness data to spend the output if they know the `secret`:
+
+| Data             | Description                                                                                             |
+|:-----------------|:--------------------------------------------------------------------------------------------------------|
+| redeem_signature | A valid SECP256k1 ECDSA DER encoded signature on the transaction with respect to the `redeem_pubkey`    |
+| redeem_pubkey    | The 33 byte SECP256k1 compressed public key that was hashed to produce the pubkeyhash `redeem_identity` |
+| secret           | The secret                                                                                              |
+| `01`             | Used to activate the redeem path in the `OP_IF`                                                         |
+| contract         | The compiled contract (as generally required when redeeming from a P2WSH output)                        |
+
+For how to use this to construct the redeem transaction see [BIP141](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#transaction-id).
+
+#### Refund
+
+To refund the HTLC, the refunder should submit a transaction to the blockchain which spends the P2WSH output.
+The refunder can use the following witness data to spend the output after the `expiry`:
+
+| Data             | Description                                                                                             |
+|:-----------------|:--------------------------------------------------------------------------------------------------------|
+| refund_signature | A valid SECP256k1 ECDSA DER encoded signature on the transaction with respect to the `refund_pubkey`    |
+| redeem_pubkey    | The 33 byte SECP256k1 compressed public key that was hashed to produce the pubkeyhash `redeem_identity` |
+| `00`              | Used to activate the refund path in the `OP_IF`                                                         |
```

Comment:
:x: The table is a little broken :construction_worker_man: 
```suggestion
| `00`             | Used to activate the refund path in the `OP_IF`                                                         |
```

Reply:
Why would deleting a space help???👷

Reply:
I'm no architect :building_construction: but it lines up all the pipes.